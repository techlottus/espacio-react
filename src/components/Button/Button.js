

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


