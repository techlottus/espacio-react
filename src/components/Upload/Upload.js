import React, { useEffect } from "react";



export const Upload = React.memo(({ data,hasError,errorMessage, onFile,onRemove }) => {
  const uploadRef = React.createRef();

  useEffect(() => {
    uploadRef.current.data = {
      title: data.title || '',
      extensions: data.extensions || '',
      disabled: data.disabled,
      id: data.id,
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    uploadRef.current.hasError = hasError;
    uploadRef.current.errorMessage = errorMessage;
  }, [hasError,errorMessage]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    uploadRef.current.addEventListener("onFile", onFile);
    uploadRef.current.addEventListener("onRemove", onRemove);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-upload ref={uploadRef}></lottus-upload>
    </>
  );
});
