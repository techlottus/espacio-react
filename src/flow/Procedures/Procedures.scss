@import "../../App.scss";

.procedures-container {
  width: 100%;
  @extend .headerapp-top;
}
