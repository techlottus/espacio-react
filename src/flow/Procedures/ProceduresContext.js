import { createContext } from "react";

export const ProceduresContext= createContext(null) 