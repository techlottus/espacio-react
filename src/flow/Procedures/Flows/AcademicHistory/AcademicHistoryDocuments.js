import React, { useContext, useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { useHistory } from "react-router";
import { Button } from "../../../../components/Button/Button";
import { Feedback } from "../../../../components/Feedback/Feedback";
import { Upload } from "../../../../components/Upload/Upload";
import {
  typeflowsProcedures,
  typeStagesProcedures,
} from "../../../../constants/Procedures.constant";
import {
  backBtnAcademicHistory,
  feedbackAcademicHistory,
  maxSizeFileInAcademicHistory,
  sendBtnAcademicHistory,
  uploadAcademicHistory,
} from "../../../../constants/ProceduresAcademicHistory.constant";
import { backStepTwoAcademicHistoryTag, uploadAcademicHistoryTag } from "../../../../tagging/flows/procedures/academicHistoryTag";
import { sendInfoTM } from "../../../../tagging/services/sendInfoTagManager";
import { ProceduresContext } from "../../ProceduresContext";
import { formatInfoModalAcademicRecordProcedure } from "../../Utils/formatInformationProcedures";
import { setInfoModalProcedure } from "../../Utils/infoModalProcedures";
import { setInfoUploadWihtoutForm } from "../../Utils/setInfoUpload";
import "./AcademicHistoryDocument.scss";

const AcademicHistoryDocuments = () => {
  const { handleSteps, handleModalProcedures, handleInfoModalProcedures } = useContext(ProceduresContext);

  const history = useHistory();

  const { texts } = useSelector((state) => state.texts);

  const [file, setFile] = useState({
    name: null,
    body: null,
    type: null,
  });
  const [next, setNext] = useState(false);
  const [errorUpload, setErrorUpload] = useState({
    hasError: false,
    errorMessage: "",
  });

  const { information: infoStore, data: dataStore } = useSelector(
    (state) => state.proceduresAcademicHistory
  );

  useEffect(() => {
    window.scrollTo(0, 0);
    if (!dataStore.phone) {
      history.replace(
        `/procedures-flows/${typeflowsProcedures.academyHistory}/${typeStagesProcedures.information}`
      );
      return;
    }
    handleSteps(1);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (next) {
      setInfoModalProcedure(
        handleModalProcedures,
        handleInfoModalProcedures,
        dataStore,
        file,
        file && file.name ? [file.name]: [],
        formatInfoModalAcademicRecordProcedure(infoStore,dataStore),
        typeflowsProcedures.academyHistory,
      )
      setNext(false);
    }
  }, [next]); // eslint-disable-line react-hooks/exhaustive-deps
  
  return (
    <>
      <div className="academicdocumentscontainer">
        <div className="forminformationacademichistory">
          <div className="academichistoryupload">
            <p className="academichistoryinfouploadtitle">
              {texts?.procedures?.uploadTextMax || ''}
            </p>
            <div className="academichistoryuploadinput">
              <Upload
                data={uploadAcademicHistory}
                hasError={errorUpload.hasError}
                errorMessage={errorUpload.errorMessage}
                onFile={(e) => {
                  setInfoUploadWihtoutForm(e.detail,setFile,setErrorUpload,maxSizeFileInAcademicHistory,texts)
                  sendInfoTM(window, uploadAcademicHistoryTag) 

                }}
                onRemove={() => {
                  setFile({
                    valid: true,
                    info: {id: uploadAcademicHistory.id, isError: true}
                  });
                }}
              />
            </div>
            <div className="academichistoryitem">
              <Feedback
                data={feedbackAcademicHistory}
                text={texts?.procedures?.feedBackText || ''}
              />
            </div>
          </div>
        </div>
        <div className="academichistorybtns">
          <div className="academicbtn">
            <Button
              data={backBtnAcademicHistory}
              onClick={() => {
                history.push(
                  `/procedures-flows/${typeflowsProcedures.academyHistory}/${typeStagesProcedures.information}`
                );
                sendInfoTM(window, backStepTwoAcademicHistoryTag) 

              }}
            />
          </div>
          <div className="academicbtn">
            <Button
              data={sendBtnAcademicHistory}
              onClick={() => {
                setNext(true);
              }}
            />
          </div>
        </div>
      </div>
    </>
  );
};

export default AcademicHistoryDocuments;
