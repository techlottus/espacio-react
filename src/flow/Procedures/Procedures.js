import React, { useEffect, useState } from "react";
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import "./Procedures.scss";
import { ProceduresContext } from "./ProceduresContext";
import {
  stepsProcedures,
  typeflowsProcedures,
} from "../../constants/Procedures.constant";
import ProcedureLayout from "./components/ProcedureLayout/ProcedureLayout";
import { ModalProcedures } from "../../components/Modal/ModalProcedures/ModalProcedures";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { proceduresRequest } from "../../middlewares/proceduresMiddlewares/indexProceduresMiddleware";
import { getProceduresCostsService } from "../../middlewares/proceduresMiddlewares/proceduresCostsMiddleware";
import { ModalAccordion } from "../../components/Modal/ModalAccordion/ModalAccordion";
import { getProfileProceduresService } from "../../middlewares/proceduresMiddlewares/procedureMainMiddleware";
import { ModalExit } from "../../components/Modal/ModalExit/ModalExit";
import { setPathExitProcedures } from "../../actions/proceduresActions/proceduresAction";
import { unblockProcedures } from "../../helpers/unblock";
import { sendInfoTM } from "../../tagging/services/sendInfoTagManager";
import {
  modalProcedureBackEquivalenceStudiesTag,
  nextStepTwoEquivalenceStudiesTag,
} from "../../tagging/flows/procedures/equivalenceStudiesTag";
import {
  modalProcedureBackAcademicHistoryTag,
  modalProcedureNextAcademicHistoryTag,
} from "../../tagging/flows/procedures/academicHistoryTag";
import { certificateStudyVerifyTag } from "../../tagging/flows/procedures/certificateStudyTag";
import {
  modalVerifyBackStudyRecordTag,
  modalVerifyNextStudyRecordTag,
} from "../../tagging/flows/procedures/studyRecordTag";
import {
  modalVerifyBackReinstatementTag,
  modalVerifyNextReinstatementTag,
} from "../../tagging/flows/procedures/reinstatementTag";
import { ModalRequirementsProcedure } from "../../components/Modal/ModalRequirementsProcedure/ModalRequirementsProcedure";
import { ModalImage } from "../../components/Modal/ModalImage/ModalImage";

const Procedures = () => {
  const [steps, setSteps] = useState({
    ...stepsProcedures,
  });

  const history = useHistory();
  const [isModal, setIsModal] = useState({
    valid: false,
    info: null,
    file: null,
    type: null,
  });

  const [isModalExit, setIsModalExit] = useState({
    isShow: false,
    unblock: null,
  });

  const [isModalAcordeon, setIsModalAcordeon] = useState(false);
  const [isModalRequirements, setIsModalRequirements] = useState({
    isShow: false,
    info: []
  });
  const [isModalImage, setIsModalImage] = useState(false);
  const [isTabs,setIsTabs] = useState(true);
  const dispatch = useDispatch();
  const { proceduresAllOne,proceduresAllTwo } = useSelector((state) => state.procedures);
  const procedures = useSelector((state) => state.procedures);

  const handleStepsBody = (focus, titles ) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
        titles
      };
    });
  };

  const handleSteps = (focus) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
      };
    });
  };

  const handleAccordion = (state) => {
    setIsModalAcordeon(state);
  };

  const handleIsTabs = (isCheck) => {
    setIsTabs(isCheck);
  }

  const handleModalRequirements = (isShow,info) => {
    setIsModalRequirements({isShow,info});
  };

  const handleModalImage = (state) => {
    setIsModalImage(state);
  };

  const handleModalProcedures = (info) => {
    setIsModal(info);
  };
  const handleRequest = () => {
    setIsModal(false);
    dispatch(
      proceduresRequest(
        isModal.type,
        history,
        isModal.info,
        isModal.file,
        isModal.detailId
      )
    );
  };

  const [infoModal, setInfoModal] = useState({
    profile: null,
    information: null,
    file: null,
    title: null,
    cost: null,
  });
  const handleInfoModalProcedures = (info) => {
    setInfoModal(info);
  };
  useEffect(() => {
    let urlSplit = history.location.pathname.split("/");

    let procedureFlow = urlSplit[2];
    let unblock = history.block((location) => {
      if (unblockProcedures(location.pathname, procedureFlow)) {
        dispatch(setPathExitProcedures(location.pathname));
        setIsModalExit({
          isShow: true,
          unblock,
        });
        return false;
      }

      return true;
    });

    if (proceduresAllOne === null || proceduresAllTwo === null) {
      dispatch(getProfileProceduresService());
      dispatch(getProceduresCostsService());
    }
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <div className="procedures-container">
        <HeaderApp />
        <ProceduresContext.Provider
          value={{
            steps,
            handleStepsBody,
            handleSteps,
            isModal,
            isTabs,
            handleModalProcedures,
            handleInfoModalProcedures,
            handleAccordion,
            handleModalRequirements,
            handleIsTabs,
            handleModalImage
          }}
        >
          <ProcedureLayout />
          <ModalExit
            isShow={isModalExit.isShow}
            onClose={() => {
              setIsModalExit((state) => {
                return {
                  ...state,
                  isShow: false,
                };
              });
            }}
            onNext={() => {
              isModalExit.unblock();
              setIsModalExit({
                isShow: false,
                unblock: null,
              });
              setPathExitProcedures(null);
              history.push(procedures.pathExit);
            }}
            flow={"procedures"}
          />
        </ProceduresContext.Provider>
        <ModalProcedures
          isShow={isModal.valid}
          outside={() => setIsModal(false)}
          onClose={() => {
            setIsModal(false);
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "back");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, modalProcedureBackEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureBackAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyBackStudyRecordTag, "back");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyBackReinstatementTag, "back");
                break;
              default:
                break;
            }
          }}
          onNext={() => {
            handleRequest();
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "next");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, nextStepTwoEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureNextAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyNextStudyRecordTag, "next");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyNextReinstatementTag, "next");
                break;
              default:
                break;
            }
          }}
          infoModal={infoModal}
        />
        <ModalAccordion
          isShow={isModalAcordeon}
          onClose={() => {
            handleAccordion(false);
          }}
          outside={() => handleAccordion(false)}
        />
        <ModalRequirementsProcedure
          isShow={isModalRequirements.isShow}
          info={isModalRequirements.info}
          onClose={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
          outside={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
        />
        <ModalImage
          isShow={isModalImage}
          onClose={() => {
            handleModalImage(false);
          }}
          outside={() => handleModalImage(false)}
        />
      </div>
    </>
  );
};

export default Procedures;
