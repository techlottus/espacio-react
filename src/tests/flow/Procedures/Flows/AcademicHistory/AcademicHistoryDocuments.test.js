import React from "react";
import { mount } from "enzyme";
import AcademicHistoryDocuments from "../../../../../flow/Procedures/Flows/AcademicHistory/AcademicHistoryDocuments";
import { Provider } from "react-redux";
import { typesRequestErrors } from "../../../../../constants/error.constant";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ProceduresContext } from "../../../../../flow/Procedures/ProceduresContext";
import { BrowserRouter } from "react-router-dom";
import { act } from "@testing-library/react";
import { Button } from "../../../../../components/Button/Button";
import {
  fileSuccessMock,
  proceduresAcademicHistoryMock,
} from "../../../../fixtures/academicHistoryDocumentsFixture";
import { Upload } from "../../../../components/Upload/Upload";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
window.scrollTo = jest.fn();

let initState = {
  proceduresAcademicHistory: {
    information: {
      optionsCampusDelivery: [],
      optionsTypeDelivery: [],
      profile: null,
      phone: null,
    },
    data: {
      selectCampusDelivery: null,
      selectTypeDelivery: null,
      comments: null,
      phone: null,
    },
    document: null,
    errors: {
      [typesRequestErrors.postAcademicHistory]: {
        isError: false,
        msg: null,
      },
    },
  },
  texts: {
    procedures: {
      uploadText: "hola",
      feedBackText: "hola",
    },
  },
};

const setSteps = jest.fn().mockImplementation(() => {});
const setIsModal = jest.fn().mockImplementation(() => {});
const setInfoModal = jest.fn().mockImplementation(() => {});
const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

const handleMount = (initState) => {
  let store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider value={{ setSteps, setIsModal, setInfoModal }}>
      <Provider store={store}>
        <BrowserRouter>
          <AcademicHistoryDocuments history={historyMock} />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <AcademicHistoryDocuments/>", () => {
  beforeEach(() => {});

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should setNext true", () => {
    initState = {
      ...initState,
      proceduresAcademicHistory: {
        ...initState.proceduresAcademicHistory,
        information: {
          ...proceduresAcademicHistoryMock.information,
        },
        data: {
          ...proceduresAcademicHistoryMock.data,
        },
      },
    };
    const wrapper = handleMount(initState);

    act(() => {
      wrapper
        .find(".academichistorybtns")
        .find(Button)
        .last()
        .props()
        .onClick();
    });
    wrapper.update();

    expect(setIsModal).toHaveBeenCalled();
    expect(setInfoModal).toHaveBeenCalled();
  });

  test("Should return to academy history info", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper
        .find(".academichistorybtns")
        .find(Button)
        .first()
        .props()
        .onClick();
    });
    wrapper.update();
  });

  describe("Should upload", () => {
    test("should come correct", async () => {
      const wrapper = handleMount(initState);
      await act(async () => {
        wrapper
          .find(Upload)
          .props()
          .onFile({
            detail: {
              ...fileSuccessMock,
            },
          });
      });
      wrapper.update();
    });
  });
});
