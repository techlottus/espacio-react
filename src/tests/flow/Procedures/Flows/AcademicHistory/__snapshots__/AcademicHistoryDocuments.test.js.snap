// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AcademicHistoryDocuments/> should init component 1`] = `
<div
  className="academicdocumentscontainer"
>
  <div
    className="forminformationacademichistory"
  >
    <div
      className="academichistoryupload"
    >
      <p
        className="academichistoryinfouploadtitle"
      />
      <div
        className="academichistoryuploadinput"
      >
        <lottus-upload />
      </div>
      <div
        className="academichistoryitem"
      >
        <lottus-feedback>
          <div
            className="feedback-content"
            content=""
          />
        </lottus-feedback>
      </div>
    </div>
  </div>
  <div
    className="academichistorybtns"
  >
    <div
      className="academicbtn"
    >
      <lottus-button />
    </div>
    <div
      className="academicbtn"
    >
      <lottus-button />
    </div>
  </div>
</div>
`;
